models/dataset.go

```diff
@@ -102,21 +102,27 @@ type ContactDetails struct {
 	Telephone string `bson:"telephone,omitempty"  json:"telephone,omitempty"`
 }
 
-// Edition represents information related to a single edition for a dataset
-type Edition struct {
-	Edition     string        `bson:"edition,omitempty"      json:"edition,omitempty"`
-	ID          string        `bson:"id,omitempty"          json:"id,omitempty"`
-	Links       *EditionLinks `bson:"links,omitempty"        json:"links,omitempty"`
-	State       string        `bson:"state,omitempty"        json:"state,omitempty"`
-	LastUpdated time.Time     `bson:"last_updated,omitempty" json:"-"`
+// EditionUpdate represents an evolving edition containing both the next and current edition
+type EditionUpdate struct {
```

Comment:
In order to be more consistent with the existing pattern for the DatasetUpdate, the Links should probably move inside the edition struct. then all the links can live in the same subdoc, and logic around getting an edition can just return everything inside the current doc without having to pull some links up a level

Reply:
corrected this, as per chat